sagemaker-debugger/debugger_interactive_analysis_profiling/interactive_analysis_profiling_data.ipynb

```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```

Comment:
<p>Do we really want to have the default value of <code>install_needed</code> be <code>False</code>? Then the onus is on the user to make sure to update the variable if their deps aren't installed yet.</p><p> </p><p>Why don't we have two cells that do something like this:</p><p> </p><pre class="ql-syntax" spellcheck="false">import sys
import IPython
import os

if os.environ.get("install_needed", "true") == "true":
&nbsp;&nbsp;&nbsp; print("installing deps and restarting kernel")
&nbsp;&nbsp;&nbsp; !{sys.executable} -m pip install -U sagemaker smdebug
&nbsp;&nbsp;&nbsp; IPython.Application.instance().kernel.do_shutdown(True)
&nbsp;&nbsp;&nbsp; install_needed = False
</pre><p> </p><pre class="ql-syntax" spellcheck="false">import os

os.environ["install_needed"] = "false"
</pre><p> </p><p>So user runs the first cell and installs deps and kernel restarts. Then running the next cell, the environment variable <code>install_needed</code> is set to False, so any execution of the first cell in this notebook or other notebooks (assuming the same environment) will not trigger a reinstall of the deps.</p><p> </p><p>Probably a simpler way to do this, but the goal is to have the user do as little as possible to run the example notebook to completion.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2174/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>